substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -797,8 +795,6 @@ impl<T: Config> Pallet<T> {
 		Self::do_remove_validator(&stash);
 		Self::do_remove_nominator(&stash);
 
-		frame_system::Pallet::<T>::dec_consumers(&stash);
```

Comment:
I would expect `assets::kill_stake` to be called here, shouldn't it?

Reply:
`StakingLedger::<T>::kill(&stash)` calls that.